File: Code Examples/JWTSecurityMongo/src/main/java/io/hashimati/security/event/LoginEventListener.java
package io.hashimati.security.event;

import io.hashimati.security.domains.LoginEvent;
import io.hashimati.security.domains.User;
import io.hashimati.security.repository.UserRepository;
import io.micronaut.context.event.ApplicationEventListener;
import jakarta.inject.Inject;
import jakarta.inject.Singleton;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Singleton
public class LoginEventListener implements ApplicationEventListener<LoginEvent>  {
    private static final Logger log = LoggerFactory.getLogger(LoginEventListener.class);

    @Inject
    private UserRepository userRepository;

    @Override
    public void onApplicationEvent(LoginEvent event) {
        log.info("Saving login event: {}", event);
       User user =  userRepository.findByUsername(event.getUsername()).block();
       user.setLastLoginStatus(event.getStatus());
       userRepository.update(user).block();
    }
}

File: Code Examples/JWT/Kotlin/JWTKotlinMongo/src/main/kotlin/io/hashimati/security/event/LoginEventListener.kt
package io.hashimati.security.event

import io.hashimati.security.domains.LoginEvent
import io.hashimati.security.repository.UserRepository
import io.micronaut.context.event.ApplicationEventListener
import jakarta.inject.Inject
import jakarta.inject.Singleton
import org.slf4j.LoggerFactory

@Singleton
class LoginEventListener(private val userRepository: UserRepository) : ApplicationEventListener<LoginEvent> {

    override fun onApplicationEvent(event: LoginEvent) {
        log.info("Saving login event: {}", event)
        val user = userRepository!!.findByUsername(event.username).block()
        user.lastLoginStatus = event.status
        user.lastTimeTryToLogin = event.lastTimeLogin
        userRepository.update(user).block()
    }

    companion object {
        private val log = LoggerFactory.getLogger(LoginEventListener::class.java)
    }
}